### ISSUE TITLE
Network special initialization

### ISSUE THREAD

**Issue by ilyanelken** [Alibaba-MIIL/ASL#13]
Hi,

I have a question regarding last layer initialisation as it was done in Focal Loss paper, so that
rare classes will initially have some low probability (Pi) of 0.01.

Do you also use some special init strategy?

Thanks,
Ilya

**Comment by mrT23** [comment]
i am not familiar with a special initialization used for Focal-Loss

if the feature layer and the last fully connected layer are i.i.d initialized (usually by xavier), almost by definition all classes
at the beginning of the training will have similar probabilities

can you refer to the part in the paper, or a known code base, that does initialization different than standard xavier ?

**Comment by ilyanelken** [comment]
It appears in Focal Loss paper in section 4.1 under "Initialization".

The classes will have the same predicted probabilities although they are not balanced in training set, right?

**Comment by mrT23** [comment]
For classification tasks, We did not do this kind of initilization, which is more relevant to detection ("anchors").
we did not see convergence issues at our training, at all

For detection training, we used mmdetection package. it can be that under the hood this initialization is used. notice that it does not depend on the loss parameters (gamma)

Tal

**Comment by ilyanelken** [comment]
I understand, thanks Tal.

Dash to Itamar from Ilya (Maxim Stepanov friend).
